go/packages/dev-mode/tests/unit/index.test.js

```diff
@@ -0,0 +1,277 @@
+'use strict';
```

Comment:
I wouldn't host any JS here, as it's outside all the lint and prettier checks, and again we duplicate all dev deps setup

Instead can we have this file introduced in otel package, so both extensions passes? I think it'll be nice to validate  two of those external extension with it

Reply:
Seems not addressed (?)

Reply:
The plan is to use the node tests while we work on rebuilding this testing framework.

Reply:
No JS code should be outside `node` folder. Can we put this as additional test to otel project, as discussed before?

Reply:
I don't know why that is would be a requirement though. These test are only related to the dev-mode extension so I don't know why I would put them in the otel folder, that feel really confusing. Especially since the plan is to replace the testing framework with something more go specific in the near future.

From my understanding the `node` folder was only created because we were creating runtime specific layers not because the code that the layer was written in was node.

Reply:
> I don't know why that is would be a requirement though.

1. All Node.js CI is configured against `node`, folder. For any JS code outside, there's no Prettier lint etc validation applied
2. All dev dependencies are configured in `node` context. Introducing any node projects outside doubles dev configuration for this repository, which should remain consistent across all work we introduce here.

So summarizing, introducing any JS packages outside makes maintenance of this repository really messy

Also, it's hard for me to understand why it's so problematic to configure this single test file in the context of otel project. 
This feels very straightforward to me, I'm happy to help if needed

Reply:
> All Node.js CI is configured against node, folder. For any JS code outside, there's no Prettier lint etc validation applied

I could always add some linting steps here if that would help.

> All dev dependencies are configured in node context. Introducing any node projects outside doubles dev configuration for this repository, which should remain consistent across all work we introduce here.

This is not using any custom node packages from this project so I am not sure what maintenance savings we would get from moving these from this folder to the other especially since these tests will be so sort lived.

It is not difficult to move the tests from this folder to the other, that is a pretty trivial task. It is just strange to have tests for one monorepo project located in another monorepo project folder.

Reply:
> I could always add some linting steps here if that would help.

No need, CI is already setup, let's simply stick to rule that no JS code lands outside of `node` folder

> It is not difficult to move the tests from this folder to the other, that is a pretty trivial task. It is just strange to have tests for one monorepo project located in another monorepo project folder.

It's only because we want to test Go project, with tests which were written for JavaScript version. I'd say that this is a _strange_ thing we're dealing with here, but we agreed it's temporary

Nonetheless, let me know, I can setup needed test file in otel project shortly.

Other benefit of having it there is that we may confirm that both external extensions pass it (that also nicely validates tests on its own with extension that was already battle-tested in prod)

Reply:
> It's only because we want to test Go project, with tests which were written for JavaScript version. I'd say that this is a strange thing we're dealing with here, but we agreed it's temporary

Yeah agreed. I think this is the is real issue here, I am almost done with converting these tests to go anyway so I will just remove these javascript tests and replace them with go tests today 👍